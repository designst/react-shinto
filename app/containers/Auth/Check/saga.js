import { call, put, select, takeLatest, getContext } from 'redux-saga/effects';

import createLogger from 'utils/createLogger';

import { makeSelectToken } from 'containers/Auth/Check/selectors';
import { CHECK_AUTH_REQUEST } from 'containers/Auth/Check/constants';
import { checkAuthSuccess, checkAuthFailure } from 'containers/Auth/Check/actions';

const logger = createLogger(__filename);

const checkAuthUrl = process.env.SHINTO_AUTH_CHECK_API_ENDPOINT;

export function* checkAuth() {
  const token = yield select(makeSelectToken());

  const apiService = yield getContext('apiService');

  logger('Check Auth: %s', token);

  try {
    const data = yield call(apiService.get, checkAuthUrl, { token });
    yield put(checkAuthSuccess(data));
  } catch (err) {
    yield put(checkAuthFailure(err));
  }
}

export default function* checkAuthData() {
  yield takeLatest(CHECK_AUTH_REQUEST, checkAuth);
}
