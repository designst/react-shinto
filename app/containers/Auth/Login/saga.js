import { call, put, takeLatest, getContext } from 'redux-saga/effects';

import createLogger from 'utils/createLogger';

import { AUTH_LOGIN_REQUEST } from 'containers/Auth/Login/constants';
import { loginSuccess, loginFailure } from 'containers/Auth/Login/actions';

const logger = createLogger(__filename);

const authLoginUrl = process.env.SHINTO_AUTH_LOGIN_API_ENDPOINT;

export function* authLogin(username, password) {
  const apiService = yield getContext('apiService');

  logger('Login: %s', username);

  try {
    const response = yield call(apiService.post, authLoginUrl, { username, password });
    yield put(loginSuccess(response.data));
  } catch (err) {
    yield put(loginFailure(err));
  }
}

export default function* authLoginData() {
  yield takeLatest(AUTH_LOGIN_REQUEST, authLogin);
}
