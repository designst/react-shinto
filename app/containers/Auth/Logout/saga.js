import { call, put, takeLatest, getContext } from 'redux-saga/effects';

import createLogger from 'utils/createLogger';

import { AUTH_LOGOUT_REQUEST } from 'containers/Auth/Logout/constants';
import { logoutSuccess, logoutFailure } from 'containers/Auth/Logout/actions';

const logger = createLogger(__filename);

const authLogoutUrl = process.env.SHINTO_AUTH_LOGOUT_API_ENDPOINT;

export function* authLogout() {
  const apiService = yield getContext('apiService');

  logger('Logout');

  try {
    const response = yield call(apiService.get, authLogoutUrl);
    yield put(logoutSuccess(response.data));
  } catch (err) {
    yield put(logoutFailure(err));
  }
}

export default function* authLogoutData() {
  yield takeLatest(AUTH_LOGOUT_REQUEST, authLogout);
}
