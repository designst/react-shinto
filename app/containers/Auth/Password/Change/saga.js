import { call, put, takeLatest, getContext } from 'redux-saga/effects';

import createLogger from 'utils/createLogger';

import { AUTH_PASSWORD_CHANGE_REQUEST } from 'containers/Auth/Password/Change/constants';

import {
  passwordChangeSuccess,
  passwordChangeFailure,
} from 'containers/Auth/Password/Change/actions';

const logger = createLogger(__filename);

const authPasswordChangeUrl = process.env.SHINTO_AUTH_PASSWORD_CHANGE_API_ENDPOINT;

export function* authPasswordChange(action) {
  const { oldPassword, newPassword, newPasswordConfirm } = action.payload;

  const apiService = yield getContext('apiService');

  logger('Password Change');

  try {
    const response = yield call(apiService.post, authPasswordChangeUrl, {
      oldPassword,
      newPassword,
      newPasswordConfirm,
    });
    yield put(passwordChangeSuccess(response.data));
  } catch (err) {
    yield put(passwordChangeFailure(err));
  }
}

export default function* authPasswordChangeData() {
  yield takeLatest(AUTH_PASSWORD_CHANGE_REQUEST, authPasswordChange);
}
