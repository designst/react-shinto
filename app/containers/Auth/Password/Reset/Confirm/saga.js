import { call, put, takeLatest, getContext } from 'redux-saga/effects';

import createLogger from 'utils/createLogger';

import { AUTH_PASSWORD_RESET_CONFIRM_REQUEST } from 'containers/Auth/Password/Reset/Confirm/constants';

import {
  passwordResetConfirmSuccess,
  passwordResetConfirmFailure,
} from 'containers/Auth/Password/Reset/Confirm/actions';

const logger = createLogger(__filename);

const authPasswordResetConfirmUrl = process.env.SHINTO_AUTH_PASSWORD_RESET_CONFIRM_API_ENDPOINT;

export function* authPasswordResetConfirm(action) {
  const { token } = action;

  const apiService = yield getContext('apiService');

  logger('Password Reset Confirm: %s', token);

  try {
    const response = yield call(apiService.post, authPasswordResetConfirmUrl, {
      token,
    });
    yield put(passwordResetConfirmSuccess(response.data));
  } catch (err) {
    yield put(passwordResetConfirmFailure(err));
  }
}

export default function* authPasswordResetConfirmData() {
  yield takeLatest(AUTH_PASSWORD_RESET_CONFIRM_REQUEST, authPasswordResetConfirm);
}
