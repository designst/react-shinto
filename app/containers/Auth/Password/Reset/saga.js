import { call, put, takeLatest, getContext } from 'redux-saga/effects';

import createLogger from 'utils/createLogger';

import { AUTH_PASSWORD_RESET_REQUEST } from 'containers/Auth/Password/Reset/constants';

import {
  passwordResetSuccess,
  passwordResetFailure,
} from 'containers/Auth/Password/Reset/actions';

const logger = createLogger(__filename);

const authPasswordResetUrl = process.env.SHINTO_AUTH_PASSWORD_RESET_API_ENDPOINT;

export function* authPasswordReset(action) {
  const { email, username } = action;

  const apiService = yield getContext('apiService');

  logger('Password Reset');

  try {
    const response = yield call(apiService.post, authPasswordResetUrl, {
      email,
      username,
    });
    yield put(passwordResetSuccess(response.data));
  } catch (err) {
    yield put(passwordResetFailure(err));
  }
}

export default function* authPasswordResetData() {
  yield takeLatest(AUTH_PASSWORD_RESET_REQUEST, authPasswordReset);
}
