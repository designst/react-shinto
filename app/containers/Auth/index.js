import { compose } from 'redux';
import { connect } from 'react-redux';

import Auth from 'components/Auth';

import injectReducer from 'utils/injectReducer';

import reducer from './reducer';

const mapStateToProps = () => ({});

const mapDispatchToProps = {};

const withReducer = injectReducer({ key: 'auth', reducer });

const withConnect = connect(mapStateToProps, mapDispatchToProps);

export default compose(withReducer, withConnect)(Auth);
