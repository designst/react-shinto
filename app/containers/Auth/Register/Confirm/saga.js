import { call, put, takeLatest, getContext } from 'redux-saga/effects';

import createLogger from 'utils/createLogger';

import { AUTH_REGISTER_CONFIRM_REQUEST } from 'containers/Auth/Register/Confirm/constants';

import {
  registerConfirmSuccess,
  registerConfirmFailure,
} from 'containers/Auth/Register/Confirm/actions';

const logger = createLogger(__filename);

const authRegisterConfirmUrl = process.env.SHINTO_AUTH_REGISTER_CONFIRM_API_ENDPOINT;

export function* authRegisterConfirm(action) {
  const { token } = action.payload;

  const apiService = yield getContext('apiService');

  logger('Register Confirm: %s', token);

  try {
    const response = yield call(apiService.post, authRegisterConfirmUrl, {
      token,
    });
    yield put(registerConfirmSuccess(response.data));
  } catch (err) {
    yield put(registerConfirmFailure(err));
  }
}

export default function* authRegisterConfirmData() {
  yield takeLatest(AUTH_REGISTER_CONFIRM_REQUEST, authRegisterConfirm);
}
