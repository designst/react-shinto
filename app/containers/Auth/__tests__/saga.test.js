import { expectSaga } from 'redux-saga-test-plan';

import authReducer from 'containers/Auth/reducer';
import { authApi } from 'containers/Auth/saga';

describe('authApiData', () => {
  let authApiGenerator;

  beforeEach(() => {
    authApiGenerator = authApi();

    const selectDescriptor = authApiGenerator.next().value;
    expect(selectDescriptor).toMatchSnapshot();
  });

  it('should dispatch the authApiSuccess action if it requests the data successfully', () =>
    expectSaga(authApi)
      .provide({
        getContext() {
          return {
            get: url => (url === '/' ? {} : {}),
          };
        },
      })
      .withReducer(authReducer)
      .hasFinalState({
        token: null,
        isAuthenticated: false,
        isAuthenticating: false,
      })
      .run());

  it('should dispatch the authApiFailure action if it requests the data unsuccessfully', () =>
    expectSaga(authApi)
      .withReducer(authReducer)
      .run());
});
