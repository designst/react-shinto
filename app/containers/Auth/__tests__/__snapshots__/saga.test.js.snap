// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`authApiData should dispatch the authApiFailure action if it requests the data unsuccessfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "GET_CONTEXT": "apiService",
}
`;

exports[`authApiData should dispatch the authApiSuccess action if it requests the data successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "GET_CONTEXT": "apiService",
}
`;
